src/debugger/debugAdapter.ts

```diff
@@ -0,0 +1,38 @@
+import { DebugAdapterTracker, DebugConsole, DebugSession } from "vscode";
+import { MessagingService } from "../service/messagingService";
+import { DEBUG_COMMANDS } from "../view/constants";
+
+export class DebugAdapter implements DebugAdapterTracker {
+    private readonly console: DebugConsole | undefined;
+    private readonly messagingService: MessagingService;
+    constructor(
+        debugSession: DebugSession,
+        messagingService: MessagingService
+    ) {
+        this.console = debugSession.configuration.console;
+        this.messagingService = messagingService;
+    }
+    onWillStartSession() {
+        // To Implement
+    }
+    onWillReceiveMessage(message: any): void {
+        if (message.command) {
+            // Only send pertinent debug messages
+            switch (message.command) {
+                case DEBUG_COMMANDS.CONTINUE:
+                    this.messagingService.sendStartMessage();
+                    break;
+                case DEBUG_COMMANDS.STACK_TRACE:
+                    this.messagingService.sendStopMessage();
```

Comment:
How is this different from `onWillStopSession()`? I think we actually want something that passes the idea that are on a broken state (e.g. `sendBreakMessage`?)

If we use "Stop" it gets the idea that we lose all our previous state.

Reply:
makes sense! I will change this. The stack trace is send from the api before entering pause or breakpoint